<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="window_size"              default="150"/>
  <arg name="frequency_tolerance"			 default="0.1"/>
  <arg name="min_acceptable_delay"     default="0"/>
  <arg name="max_acceptable_delay"		 default="0.2"/>
  <arg name="update_interval"          default="1"/>
  <arg name="min_frequency"			 			 default="15"/>
  <arg name="max_frequency"			 			 default="15"/>
  <arg name="use_sim_time"             default="false"/>

  <arg name="data_topics"              default="[/k1/hd/camera_info,
                                                 /k2/hd/camera_info,
                                                 /k3/hd/camera_info,
                                                 /k4/hd/camera_info,
                                                 /k5/hd/camera_info,
                                                 /k1/sd/camera_info,
                                                 /k2/sd/camera_info,
                                                 /k3/sd/camera_info,
                                                 /k4/sd/camera_info,
                                                 /k5/sd/camera_info]"/>

  <group ns="diagnostics">
		<node pkg="gear_diagnostics" type="camera_monitor" name="$(anon camera_diagnostics)" output="screen">
      <rosparam param="data_topics" subst_value="True">$(arg data_topics)</rosparam>
		  <param name="window_size"              	value="$(arg window_size)"/>
		  <param name="frequency_tolerance"			 	value="$(arg frequency_tolerance)"/>
		  <param name="min_acceptable_delay"     	value="$(arg min_acceptable_delay)"/>
		  <param name="max_acceptable_delay"		 	value="$(arg max_acceptable_delay)"/>
		  <param name="update_interval"          	value="$(arg update_interval)"/>
		  <param name="min_frequency"			 			 	value="$(arg min_frequency)"/>
		  <param name="max_frequency"			 			 	value="$(arg max_frequency)"/>
		  <param name="use_sim_time"  						value="$(arg use_sim_time)"/>
		</node>
	</group>
</launch>